app/controllers/search.controller.js

```diff
@@ -1,55 +1,136 @@
+const cron = require('node-schedule');
+const puppeteer = require('puppeteer-extra');
+const { Op } = require('sequelize');
+const StealthPlugin = require('puppeteer-extra-plugin-stealth');
+
 const db = require('../models');
+const KeywordManager = require('../event-emitters/keywordsManager');
+const timer = require('../utils/timer');
 
-const puppeteer = require('puppeteer');
-const { Op } = require('sequelize');
+// Bypass mass-searching Google (sometimes it works, and sometimes it's not)
+puppeteer.use(StealthPlugin());
```

Comment:
A more reliable way is to rotate the user agent for each request. This would allow passing through most checks as each request would be identified as a different user device.

Reply:
actually, I tried to use the package user-agent but it didn't work as expected so that's why I change to this npm package.